### PR TITLE
feat: enable config checker

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
References https://github.com/akka/akka-diagnostics/issues/19

### CODE REVIEW

**Comment by franciscolopezsancho** [comment]
I added the config checker to all of them but I could only prove it works in the `shopping-cart-service` applications. Java and Scala. 
Couldn't prove it works in the `shopping-analytics-service` because these two applications don't start:
```
[info] [2023-01-31 17:48:20,468] [WARN] [akka.stream.scaladsl.RestartWithBackoffSource] [akka://ShoppingAnalyticsService@127.0.0.1:3551] [] [ShoppingAnalyticsService-akka.actor.default-dispatcher-18] - Restarting stream due to failure [4]: java.lang.RuntimeException: Failure to consume gRPC event stream for [cart]
[info] java.lang.RuntimeException: Failure to consume gRPC event stream for [cart]
[info] 	at akka.projection.grpc.consumer.scaladsl.GrpcReadJournal$$anonfun$1.applyOrElse(GrpcReadJournal.scala:262)
[info] 	at akka.projection.grpc.consumer.scaladsl.GrpcReadJournal$$anonfun$1.applyOrElse(GrpcReadJournal.scala:260)
[info] 	at akka.stream.impl.fusing.Recover$$anon$7.onUpstreamFailure(Ops.scala:295)
[info] 	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:525)
[info] 	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:390)
[info] 	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:650)
[info] 	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:521)
[info] 	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:625)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:800)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$shortCircuitBatch(ActorGraphInterpreter.scala:787)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:819)
[info] 	at akka.actor.Actor.aroundReceive(Actor.scala:537)
[info] 	at akka.actor.Actor.aroundReceive$(Actor.scala:535)
[info] 	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:716)
[info] 	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:579)
[info] 	at akka.actor.ActorCell.invoke(ActorCell.scala:547)
[info] 	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
[info] 	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
[info] 	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
[info] 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
[info] 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[info] 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[info] 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[info] 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[info] Caused by: akka.grpc.GrpcServiceException: UNAVAILABLE: io exception
[info] 	at akka.grpc.GrpcServiceException$.apply(GrpcServiceException.scala:53)
[info] 	at akka.grpc.internal.RequestBuilderImpl$.lift(RequestBuilderImpl.scala:407)
[info] 	at akka.grpc.internal.RequestBuilderImpl$$anonfun$richErrorStream$1.applyOrElse(RequestBuilderImpl.scala:399)
[info] 	at akka.grpc.internal.RequestBuilderImpl$$anonfun$richErrorStream$1.applyOrElse(RequestBuilderImpl.scala:398)
[info] 	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:35)
[info] 	at akka.stream.impl.fusing.RecoverWith$$anon$43.akka$stream$impl$fusing$RecoverWith$$anon$$onFailure(Ops.scala:2152)
[info] 	at akka.stream.impl.fusing.RecoverWith$$anon$43.onUpstreamFailure(Ops.scala:2148)
[info] 	... 21 more
[info] Caused by: io.grpc.netty.shaded.io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:8101
[info] Caused by: java.net.ConnectException: Connection refused
[info] 	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[info] 	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[info] 	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[info] 	at io.grpc.netty.shaded.io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
[info] 	at io.grpc.netty.shaded.io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
[info] 	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:710)
[info] 	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658)
[info] 	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584)
[info] 	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
[info] 	at io.grpc.netty.shaded.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995)
[info] 	at io.grpc.netty.shaded.io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[info] 	at io.grpc.netty.shaded.io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[info] 	at java.base/java.lang.Thread.run(Thread.java:833)
^C[info] [2023-01-31 17:48:39,009] [INFO] [akka.actor.CoordinatedShutdown] [akka://ShoppingAnalyticsService@127.0.0.1:3551] [] [ShoppingAnalyticsService-akka.actor.default-dispatcher-18] - Running CoordinatedShutdown with reason [JvmExitReason]
[info] [2023-01-31 17:48:39,014] [INFO] [akka.cluster.sharding.DDataShardCoordinator] [akka://ShoppingAnalyticsService@127.0.0.1:3551] [] [ShoppingAnalyticsService-akka.actor.default-dispatcher-3] - sharded-daemon-process-cart-events: Starting shutting down shards [0,1,2,3] due to region shutting down.
```

File: samples/grpc/shopping-analytics-service-scala/build.sbt [hunk 1]
```diff
@@ -28,11 +28,12 @@ run / javaOptions ++= sys.props
 Global / cancelable := false // ctrl-c
 
 val AkkaVersion = "2.7.0"
-val AkkaHttpVersion = "10.4.0"
+val AkkaHttpVersion = "10.5.0-M1"
```
[the remark this answers is not in the document]
Comment (johanandren) [reply samples/grpc/shopping-analytics-service-scala/build.sbt:31]: Hmm, curious, we don't use `akka-http-spray-json` and `akka-http2-support` only contains tests, would be good to look at the dependency graph and see if you accidentally pulled those in through diagnostics (it shouldn't have any dependencies, other than akka core, right?)

**Comment by johanandren** [comment]
To run the analytics samples you'd need to start the cart first as they consumes the gRPC stream from those, if not available they will keep hitting the configured endpoint to try to connect and throw/log an exception each time. However, the config checker runs on ActorSystem start so running it successful after that is not that interesting.

**Comment by franciscolopezsancho** [comment]
I see. Thanks for the info 👍 . Now that I'm here I think I could add your insight in the `README.md` so the next one understands the logs right away.

**Comment by octonato** [comment]
why 19?? was that a mistake?

**Comment by franciscolopezsancho** [comment]
> why 19?? was that a mistake?

It's a new project

File: samples/grpc/shopping-analytics-service-scala/build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (franciscolopezsancho) [reply samples/grpc/shopping-analytics-service-scala/build.sbt:31]: Okay. I'll have a look

**Comment by patriknw** [comment]
It references https://github.com/akka/akka-diagnostics/issues/19 but that is somewhat misleading to have in a commit message of another repository, but not a big deal.

**Comment by franciscolopezsancho** [comment]
> It references [akka/akka-diagnostics#19](https://github.com/akka/akka-diagnostics/issues/19) but that is somewhat misleading to have in a commit message of another repository, but not a big deal.

Oh, the title. Fixed

**Comment by patriknw** [comment]
Also enable it in the replicated event sourcing samples? https://github.com/akka/akka-projection/tree/main/samples/replicated

File: samples/grpc/shopping-analytics-service-scala/build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (franciscolopezsancho) [reply samples/grpc/shopping-analytics-service-scala/build.sbt:31]: Unsurprisingly you were correct :) . I removed the unnecessary dependencies from `akka-diagnostics` now I will need to publish the new version.

**Comment by franciscolopezsancho** [comment]
Back to draft until I release `akka-diagnostics` '2.0.0-M3'

**Review by johanandren — APPROVED** [review]

**Pull request merged**